### PR TITLE
search: if not specified, set fork:no by default

### PR DESCRIPTION
**Pull request by rvantonder**
Stacked on #8737. Product approves, we will exclude forks default. See #7004 for related issue.

Will add e2e test if approved before merging.

### CODE REVIEW

**Comment by rvantonder** [comment]
Agree see https://github.com/sourcegraph/sourcegraph/pull/8737#issuecomment-594212739

**Comment by rvantonder** [comment]
Added the alert suggestion.

Screen Shot 2020-03-09 at 2 29 10 PM

File: cmd/frontend/graphqlbackend/search_alert.go [hunk 1]
```diff
@@ -288,6 +289,21 @@ func (r *searchResolver) alertForNoResolvedRepos(ctx context.Context) *searchAle
 		}
 
 		proposedQueries := []*searchQueryDescription{}
+		if forksNotSet {
+			tryIncludeArchived := resolveRepoOp{
+				repoFilters:      repoFilters,
+				minusRepoFilters: minusRepoFilters,
+				noForks:          false,
+			}
+			if reposExist(ctx, tryIncludeArchived) {
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/search_alert.go]:
```suggestion
		if forksNotSet {
			tryIncludeForks := resolveRepoOp{
				repoFilters:      repoFilters,
				minusRepoFilters: minusRepoFilters,
				noForks:          false,
			}
			if reposExist(ctx, tryIncludeForks) {
```

Comment (rvantonder) [reply cmd/frontend/graphqlbackend/search_alert.go]: Thanks for the catch @eseliger! I award you 12 internet points.

**Pull request merged**